velox/exec/RowContainer.h

```diff
@@ -817,13 +841,22 @@ class RowContainer {
   /// invalidated. Any row erase operations will invalidate column stats.
   std::optional<RowColumn::Stats> columnStats(int32_t columnIndex) const;
 
+  uint32_t columnNullCount(int32_t columnIndex) {
+    return rowColumnsStats_[columnIndex].nullCount();
+  }
+
+  bool isMinMaxColumnStatsValid() const {
+    return rowColumnsStatsMinMaxValid_;
+  }
+
   const auto& keyTypes() const {
     return keyTypes_;
   }
 
-  /// Returns true if specified column may have nulls, false otherwise.
+  /// Returns true if specified column has nulls, false otherwise.
   inline bool columnHasNulls(int32_t columnIndex) const {
-    return columnHasNulls_[columnIndex];
+    return columnStats(columnIndex).has_value() &&
```

Comment:
since we always has column stats now so can we change columnStats() to  const RowColumn::Stats& RowContainer::columnStats? thanks!

Reply:
It's unlikely but is it possible that it is initialized with empty rows?

Reply:
The nullopt is to indicate invalidated column stats. Now the invalidation will be of finer granularity, So the column stats for each column will always be valid (not nullopt) but just some fields inside column stats invalidated under some conditions.